### PR TITLE
Prevent an integer overflow when checking start_func_index

### PR DESCRIPTION
**Pull request by lum1n0us**
[69920](https://oss-fuzz.com/testcase-detail/5082950006276096)

### CODE REVIEW

File: core/iwasm/aot/aot_loader.c [hunk 1]
```diff
@@ -2486,7 +2486,7 @@ load_init_data_section(const uint8 *buf, const uint8 *buf_end,
     /* check start function index */
     if (module->start_func_index != (uint32)-1
         && (module->start_func_index
-            >= module->import_func_count + module->func_count)) {
+            >= (uint64)module->import_func_count + module->func_count)) {
```
Comment (yamt) [inline core/iwasm/aot/aot_loader.c:2489]: the overflow check should be done earlier, regardless of start_func_index.

**Review by yamt — CHANGES_REQUESTED** [review]

**Comment by yamt** [comment]
btw, does it make much sense to perform fuzzing on the aot loader? (i guess it's what you are doing. this is just a guess because i have no access to the given oss-fuzz url.)
unlike wasm modules, we somehow need to trust aot modules in some extent anyway.

File: core/iwasm/aot/aot_loader.c (hunk 1 above)
Comment (wenyongh) [reply core/iwasm/aot/aot_loader.c:2489]: yes, maybe we can check the integer overflow after reading `module->func_count`

**Comment by lum1n0us** [comment]
Use https://github.com/bytecodealliance/wasm-micro-runtime/pull/3579 to fix

**Pull request closed without merging**